tap_exacttarget/dao.py

```diff
@@ -36,9 +38,31 @@ def load_schema(stream):
 
     return schema
 
+# boolean function to check if the error is 'timeout' error or not
+def is_timeout_error():
+    """
+        This function checks whether the URLError contains 'timed out' substring and return boolean
+        values accordingly, to decide whether to backoff or not.
+    """
+    def gen_fn(exc):
```

Comment:
@harshpatel4crest Check if having on `gen_fn()` works for timeout

Reply:
Added a single function for giveup condition.